charts/redpanda/templates/_configmap.tpl

```diff
@@ -403,61 +403,144 @@ limitations under the License.
-{{- end }}
-{{- if hasKey .Values.tuning "well_known_io" }}
-      well_known_io: {{ .Values.tuning.well_known_io }}
-{{- end }}
+{{- define "rpk-config-internal" -}}
+  {{- $brokers := list -}}
+  {{- $admin := list -}}
+  {{- range $i := untilStep 0 (.Values.statefulset.replicas|int) 1 -}}
+    {{- $podName := printf "%s-%d.%s" (include "redpanda.fullname" $) $i (include "redpanda.internal.domain" $) -}}
+    {{- $brokers = concat $brokers (list (printf "%s:%d" $podName (int $.Values.listeners.kafka.port))) -}}
+    {{- $admin = concat $admin (list (printf "%s:%d" $podName (int $.Values.listeners.admin.port))) -}}
+  {{- end -}}
+rpk:
+  # redpanda server configuration
+  overprovisioned: {{ dig "cpu" "overprovisioned" false .Values.resources }}
+  enable_memory_locking: {{ dig "memory" "enable_memory_locking" false .Values.resources }}
+  additional_start_flags:
+    - "--smp={{ include "redpanda-smp" . }}"
+    - "--memory={{ template "redpanda-memory" . }}M"
+    - "--reserve-memory={{ template "redpanda-reserve-memory" . }}M"
+    - "--default-log-level={{ .Values.logging.logLevel }}"
+  {{- with .Values.statefulset.additionalRedpandaCmdFlags -}}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+
+  {{- with dig "config" "rpk" dict .Values.AsMap }}
+  # config.rpk entries
+  {{- toYaml . | nindent 2 }}
+  {{- end }}
+
+  {{- with dig "tuning" dict .Values.AsMap }}
+  # rpk tune entries
+  {{- toYaml . | nindent 2 }}
+  {{- end }}
+
+  # kafka connection configuration
+  kafka_api:
+    brokers: {{ toYaml $brokers | nindent 6 }}
+    tls:
+  {{- if (include "tls-enabled" . | fromJson).bool }}
+    {{- $cert := get .Values.tls.certs .Values.listeners.kafka.tls.cert }}
+    {{- if $cert.caEnabled }}
+      truststore_file: {{ printf "/etc/tls/certs/%s/ca.crt" .Values.listeners.kafka.tls.cert }}
+    {{- end }}
+    {{- if .Values.listeners.kafka.tls.requireClientAuth }}
+      cert_file: {{ printf "/etc/tls/certs/%s-client/tls.crt" (include "redpanda.fullname" .) }}
+      key_file: {{ printf "/etc/tls/certs/%s-client/tls.key" (include "redpanda.fullname" .) }}
+    {{- end }}
+  {{- end }}
+  admin_api:
+    addresses: {{ toYaml $admin | nindent 6 }}
+    tls:
+  {{- if (include "tls-enabled" . | fromJson).bool }}
+    {{- $cert := get .Values.tls.certs .Values.listeners.admin.tls.cert }}
+    {{- if $cert.caEnabled }}
+      truststore_file: {{ printf "/etc/tls/certs/%s/ca.crt" .Values.listeners.admin.tls.cert }}
+    {{- end }}
+    {{- if .Values.listeners.admin.tls.requireClientAuth }}
+      cert_file: {{ printf "/etc/tls/certs/%s-client/tls.crt" (include "redpanda.fullname" .) }}
+      key_file: {{ printf "/etc/tls/certs/%s-client/tls.key" (include "redpanda.fullname" .) }}
+    {{- end }}
+  {{- end }}
 {{- end -}}
 
 {{- define "configmap-server-list" -}}
-{{- $root := . }}
-{{- range (include "seed-server-list" $root | mustFromJson) }}
+  {{- range (include "seed-server-list" . | mustFromJson) }}
 - host:
```

Comment:
Is this -host list indentation correct? I believe it at the `rpk:` level which is in my opinion wrong place.